lib/tpm2_eventlog_yaml.c

```diff
@@ -413,85 +440,93 @@ bool yaml_uefi_image_load(UEFI_IMAGE_LOAD_EVENT *data, size_t size) {
+                         header->Revision,
+                         header->HeaderSize,
+                         header->HeaderCRC32,
+                         header->MyLBA,
+                         header->AlternateLBA,
+                         header->FirstUsableLBA,
+                         header->LastUsableLBA,
+                         guid,
+                         header->PartitionEntryLBA,
+                         header->NumberOfPartitionEntries,
+                         header->SizeOfPartitionEntry,
+                         header->PartitionEntryArrayCRC32,
+                         data->NumberOfPartitions);
+
+        size -= (sizeof(data->UEFIPartitionHeader) + sizeof(data->NumberOfPartitions));
+
+        UINT64 i;
+        UEFI_PARTITION_ENTRY *partition = data->Partitions;
+        for (i = 0; i < data->NumberOfPartitions; i++) {
+            if (size < sizeof(*partition)) {
+                LOG_ERR("Cannot parse GPT partition entry: insufficient data (%zu)\n", size);
+                return false;
+            }
+
+            uuid_unparse_lower(partition->PartitionTypeGUID, guid);
+            tpm2_tool_output("      - PartitionTypeGUID: %s\n", guid);
+            uuid_unparse_lower(partition->UniquePartitionGUID, guid);
+            tpm2_tool_output("        UniquePartitionGUID: %s\n"
+                             "        StartingLBA: 0x%" PRIx64 "\n"
+                             "        EndingLBA: 0x%" PRIx64 "\n"
+                             "        Attributes: 0x%" PRIx64 "\n"
+                             "        PartitionName: \"%.*s\"\n",
+                             guid,
+                             partition->StartingLBA,
+                             partition->EndingLBA,
+                             partition->Attributes,
+                             72, partition->PartitionName);
+            size -= sizeof(*partition);
         }
 
-        uuid_unparse_lower(partition->PartitionTypeGUID, guid);
-        tpm2_tool_output("      - PartitionTypeGUID: %s\n", guid);
-        uuid_unparse_lower(partition->UniquePartitionGUID, guid);
-        tpm2_tool_output("        UniquePartitionGUID: %s\n"
-                         "        StartingLBA: 0x%" PRIx64 "\n"
-                         "        EndingLBA: 0x%" PRIx64 "\n"
-                         "        Attributes: 0x%" PRIx64 "\n"
-                         "        PartitionName: \"%.*s\"\n",
-                         guid,
-                         partition->StartingLBA,
-                         partition->EndingLBA,
-                         partition->Attributes,
-                         72, partition->PartitionName);
-        size -= sizeof(*partition);
+        if (size != 0) {
+            LOG_ERR("EventSize is inconsistent with actual data\n");
+            return false;
+        }
     }
-
-    if (size != 0) {
-        LOG_ERR("EventSize is inconsistent with actual data\n");
-        return false;
+    else {
```

Comment:
else on same line as closing of related if

Reply:
yep, will fix